inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/BinFunction.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.process.function;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.Function;
+
+import java.math.BigDecimal;
+import java.util.List;
+/**
+ * BinFunction
+ * description: bin(integer)--Returns a string representation of an integer in binary format. If the integer is NULL, NULL is returned.
+ */
+public class BinFunction implements ValueParser {
+
+    private ValueParser valueParser;
+
+    public BinFunction(Function expr) {
+        if (expr.getParameters() != null) {
+            List<Expression> expressions = expr.getParameters().getExpressions();
+            if (expressions != null && expressions.size() == 1) {
+                valueParser = OperatorTools.buildParser(expressions.get(0));
+            }
+        }
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        if (valueParser != null) {
+            Object valueObj = valueParser.parse(sourceData, rowIndex, context);
```

Comment:
is it possible that `valueObj` is null?

Reply:
> is it possible that `valueObj` is null?

I think it’s no possible, `valueParser` has been checked in constructor and it can’t be null here, so `valueObj` cannot be null here too. Should I verify the existence of valueObj here?

Reply:
No need to do a duplicate check